### ISSUE TITLE
Hang inside Docker container

### ISSUE THREAD

**Issue by jtyers** [googleapis/nodejs-pubsub#686]
I have a `docker-compose.yml` file that runs integration tests, which includes talking to Pub/Sub. It looks like this:

```
version: "3.4"
services:
  app:
    image: node:10-alpine
    volumes:
    - ./:/app:ro
    working_dir: /app
    environment:
      - GOOGLE_APPLICATION_CREDENTIALS=/app/service_account.json
    entrypoint:
      - npm
      - run
      - test-e2e-debug
```

The tests hang when the call to `publish()` takes place. The relevant code:

```js
exports = module.exports = async (data) => {
  const dataBuffer = Buffer.from(JSON.stringify(data));

  const topic = pubsub.topic(topicName, {
    batching: {
     maxMessages: 1, // max messages per batch
     maxMilliseconds: 50, // max wait time (ms) per batc
    },
  });

  const { publisher } = topic;
  const messageId = await publisher.publish(dataBuffer);
  log.error({ messageId }, 'pubsub complete');

  return messageId;
};
```

Running the same function **outside** the container, directly on my machine, works without issue. I'm using service account JSON creds with `GOOGLE_APPLICATION_CREDENTIALS` set in my environment, both in the container and outside.

Incidentally, when I configured `GOOGLE_APPLICATION_CREDENTIALS` to a nonexistent filename, it did cause an error to be thrown, so I assume the correct filename I've now set is being picked up correctly.

I've done a little debugging inside the pubsub client and can confirm that `publish_()` is called, but then it simply hangs. Running it in the debugger with 'pause on caught exceptions' causes no break. The request callback in `publish_()` never gets fired.

So my issue is two-fold:
1. why the hang inside my container, rather than chucking out an error somewhere? (is there a retry-backoff loop going on?)
2. how do I correctly configure authentication when running inside a container, given service account JSON credentials?

**Comment by JeremyPlease** [comment]
I'm having the same issue here. This minimal code reproduces it:

Docerkfile:
```
FROM node:8.15.0-alpine
CMD ["echo", "Done!"]
```

Test script:
```
const { PubSub } = require('@google-cloud/pubsub');
const pubsubClient = new PubSub();
pubsubClient.topic('testing').publish(Buffer.from('test')).then(console.log, console.error)
```
Nothing is logged and it just hangs forever.

**Comment by JeremyPlease** [comment]
Here are some simplified steps for reproducing:

1. Open ssh of docker image: `docker run -ti --entrypoint=sh node:8.15.0-alpine`
1. Install `npm i '@google-cloud/pubsub'`
1. Start nodejs shell: `node`
1. Paste in this code:
```
const { PubSub } = require('@google-cloud/pubsub');
const pubsubClient = new PubSub();
pubsubClient.topic('testing').publish(Buffer.from('test')).then(console.log, console.error)
setInterval(() => console.log('in interval'), 500);
```

"in interval" will print ~3 times and then it freezes and node shell can't be escaped.

**Comment by callmehiphop** [comment]
It's been a minute, but there hasn't really been any activity here lately. Has anyone been experiencing this issue still?

**Comment by JeremyPlease** [comment]
Thanks for checking in. It looks like this was resolved in some version after v0.30.0. It now properly throws an `Unable to detect a Project Id in the current environment.`error instead of locking. This issue can be closed now.

**Comment by callmehiphop** [comment]
Awesome, thanks for the update @JeremyPlease!
